Assets/Scripts/Resources/ResourceSpawner.cs

```diff
@@ -11,30 +11,76 @@ public class ResourceSpawner : MonoBehaviour
 	public int total;
 	public float radius;
 
+	private static int maxResourceSpawnTries = 100;
+
+	private new Transform transform = null;
+
 	void Start()
 	{
 		// configure resource holder for the current resource type
 		resourceHolderPrefab.resource = resourcePrefab;
 
+		transform = GetComponent<Transform>();
+
+		if ( !resourcePrefab.GetComponent<Collider>() )
```

Comment:
Have the `resourcePrefab` member be of type `Collider`, then you won't have to do this check.